src/components/Main/NewSurveyPage/NewSurveyPage.component.jsx

```diff
@@ -14,18 +16,23 @@ const NewSurveyPage = props => {
     indicator: false
   };
 
-  const [questionState, addNewQuestion] = useState([]);
+  const [surveyState, updateSurveyState] = useState({
+    name: '',
+    id: 's-header',
+    questions: []
+  });
+  const [questionState, updateQuestionState] = useState({});
   const [inputText, updateInputText] = useState({});
 
   const handleQuestionTypeClick = e => {
     e.preventDefault();
-    const question = cloneDeep(questionState);
+    let question = cloneDeep(questionState);
     cloneDeep(props.surveyQuestionTypes).map(q => {
       if (q.name === e.target.innerText) {
-        question[0] = q;
+        question = q;
```

Comment:
Not clear why you need cloneDeep(props.surveyQuestionTypes)?
If the goal is to find question from props and if it's not found, give it a default state, it can be done dirrefently

Reply:
import surveyQuestionTypes = [] from constants;
const type = surveyQuestionTypes.find(({name}) => name === target.value)
const q = {...cloneDeep(state), ...type};

Reply:
done, refactored this part